### PR TITLE
refactor: Consistent G2Point with arkworks

### LINKED ISSUE #4: G2Point definition inconsistent with arkwork
Our current G2Point definition is:

```solidity
    // G2 group element where x \in Fp2 = x0 * z + x1
    struct G2Point {
        uint256 x0;
        uint256 x1;
        uint256 y0;
        uint256 y1;
    }
```

Whereas in [arkworks](https://docs.rs/ark-ff/latest/ark_ff/fields/models/quadratic_extension/struct.QuadExtField.html):
```rust
pub struct QuadExtField<P: QuadExtConfig> {
    /// Coefficient `c0` in the representation of the field element `c = c0 + c1 * X`
    pub c0: P::BaseField,
    /// Coefficient `c1` in the representation of the field element `c = c0 + c1 * X`
    pub c1: P::BaseField,
}
```

meaning `G2Point.x0 = QuadExtField.c1` and `G2Point.x1 = QuadExtField.c0`, this is very error prone for the downstream users!

This issue aims to reconcile the inconsistency even though it's not an error and changes would be breaking.

cc @sveitser @philippecamacho

### PR DESCRIPTION
**Pull request by alxiong**
Closes #4 

_dep: should be merged after #5_

### CODE REVIEW

File: diff-test/src/main.rs [hunk 1]
```diff
@@ -31,6 +38,40 @@ fn main() {
...
+            let alpha_r = Fr::rand(&mut rng);
+            let beta_r = alpha_l * beta_l / alpha_r;
+
+            let a_1 = G1Affine::generator() * alpha_l;
+            let a_2 = G2Affine::generator() * beta_l;
+            let b_1 = G1Affine::generator() * alpha_r;
+            let mut b_2 = G2Affine::generator() * beta_r;
+
+            assert_eq!(Bn254::pairing(a_1, a_2), Bn254::pairing(b_1, b_2));
+            if seed % 2 == 0 {
```
Comment (philippecamacho) [inline diff-test/src/main.rs:63]: Why do we need this condition?

Comment (alxiong) [reply diff-test/src/main.rs:63]: I want to test both positive and negative paths. I can either split the tests into two test, or depending on the value of `seed`, choose to generate "satisfying pairs" or "unsatisfying pairs"

Comment (philippecamacho) [reply diff-test/src/main.rs:63]: I see thanks.

**Review by philippecamacho — APPROVED** [review]
Thanks for that, I lost quite some time due to this inconsistency :)

**Pull request merged**